### DISCUSSION TITLE
Belchertown MQTT with HA Broker

### DISCUSSION THREAD

**Discussion by damclarkd** [poblabs/weewx-belchertown#781] [Q&A] [answered]
I have been using the Belchertown skin on weewx with a MQTT broker in Home Assistant for some time now. All working good. Now the MQTT broker addin in HA has been updated and anonymous logins are no longer supported. I have added a user and password in the broker and into my weewx.conf with mqtt_websockets_username and mqtt_websockets_password fields but connections are refused. "null unpwd" in the broker logs. I am not using SSL

Are there different credentials or configurations needed for an updated Mosquitto? Anyone have this working with the Home Assistant broker?

**Comment by ThePeePs** [comment]
what versions of the skin and mosquitto are you using?

**Reply by damclarkd** [reply]
Well I am troubleshooting the issue now and thinking I had the latest version 1.2 of the skin. However I find that the version of belchertown.js.tmpl does not have the mqtt_websockets_password nor mqtt_websockets_username details. So guessing that is an older version. The Home Assistant Mosquitto Broker Addin is at version 2.0.11 of mosquitto. 
The data is being published to the broker correctly by passing the username and password in the URL. However getting the data via websockets to update the website live is failing with authentication issues.
Trying to sort how best to proceed to get all the updated files for the skin. Wipe out all Belchertown Skin from system and re-install??

**Reply by damclarkd** [reply]
Skin WAS at version 1.2. However some of the files like belchertown.js do seem to be older. Not sure why there are older files in the latest release?

**Comment by ThePeePs** [comment, marked as answer]
Version 1.2 of the skin is the current stable release.  If you want to use 1.3b1, you would need to clone the current main branch and then manually set it up in weewx.  I know you said  that you aren't using SSL, but I did come across a [bug in 2.0.11 of Mosquitto](https://github.com/poblabs/weewx-belchertown/discussions/777) that affects SSL, I'm not sure if it also affects unot setting a user/password as well (I don't have them set on my setup).  I have noticed that having blank, or null username and password (ie: user = "") set for seems to break the websocket connection.  I'm unsure if this is a bug in Mosquitto or in the JS websocket client.

**Reply by damclarkd** [reply]
Sounds like the path I will need to go after. The un/pw will be needed in my use case, I will likely take this time to also set up SSL for this connection. Thanks for the response.
Cheers

**Reply by damclarkd** [reply]
Many thanks again. Up and running with the master version and MQTT fully implemented and updating.
Cheers!
